src/analyze.cpp

```diff
@@ -5458,8 +5458,19 @@ bool const_values_equal(ConstExprValue *a, ConstExprValue *b) {
         case TypeTableEntryIdPointer:
         case TypeTableEntryIdFn:
             return const_values_equal_ptr(a, b);
-        case TypeTableEntryIdArray:
-            zig_panic("TODO");
+        case TypeTableEntryIdArray: {
+            assert(a->type->data.array.len == b->type->data.array.len);
+            size_t len = a->type->data.array.len;
+            ConstExprValue *a_elems = a->data.x_array.s_none.elements;
```

Comment:
I think before you access `s_none` you should assert that the array is not `ConstArraySpecialUndef`

This is an optimization so that if the user has `var blah: [2 * 1024 * 1024 * 1024]u8 = undefined`, we don't bother allocating 2 GiB of memory at compile time. Probably before we invoke `const_values_equal` for arrays, we should either expand the undef (there's a function for that somewhere), or emit a compile error for trying to do an equality comparison with undefined at compile-time. Seems like the latter would be more appropriate.

Let me know if you need help coming up with a test case for this.